### ISSUE TITLE
Plan to implement more data augmentation methods, need some suggestions

### ISSUE THREAD

**Issue by irvingzhang0512** [open-mmlab/mmaction2#436]
**Describe the feature**
More data augmentation methods.

**Motivation**
When training our own models, we need to TRY EVERYTHING...

**Additional context**
+ MMaction2 existing data augmentation methods:
    + random flip.
    + all kinds of crop and resize
    + color jitter(brightness/contrast/saturation/hue)

**Plans**
+ [ ] Support third party data augmentation library in pipeline.
    + plan to support [imgaug](https://github.com/aleju/imgaug) and [albumentations](https://github.com/albumentations-team/albumentations)
+ [ ] Support `mixup`
    + [paper](https://arxiv.org/abs/1710.09412) and [repo](https://github.com/facebookresearch/mixup-cifar10)
+ More image augmentation methods could refer to [PaddleClas](https://github.com/PaddlePaddle/PaddleClas/blob/master/docs/en/advanced_tutorials/image_augmentation/ImageAugment_en.md).

**Comment by dreamerlin** [comment]
You can take a PR for it and we will help to review it : )

**Comment by innerlee** [comment]
Thanks for the enthusiastic! Here is a practical way:

1. Make a list of candidate augmentations. For each augmentation, find its original paper, together with a reference implementation.
2. Make individual pull requests for each one of the list.

You may use this issue to track progress.

**Comment by dreamerlin** [comment]
Yeah, Thanks for your proposal. It is better to provide a list of candidate augmentations so that we can also help to implement some of these ! @irvingzhang0512

**Comment by irvingzhang0512** [comment]
[this paper](https://arxiv.org/abs/2012.06567) shows that random rotation may help
![image](https://user-images.githubusercontent.com/22089207/102425563-86e66080-4048-11eb-88e4-344ed19cc01d.png)

**Comment by innerlee** [comment]
In image classification, rotation works to some extent. But color jittering sometimes does not work.

Whether they work or not, it does not affect whether we implement them.

**Comment by innerlee** [comment]
Supporting rotation would be great.

I also wanted to have some camera transformations, such as 
* push in
* pull out
* pan
* tilt

Do you have interest in implementing them?

**Comment by irvingzhang0512** [comment]
> Supporting rotation would be great.
> 
> I also wanted to have some camera transformations, such as
> 
> * push in
> * pull out
> * pan
> * tilt
> 
> Do you have interest in implementing them?

@innerlee 
I'm a little busy until Jan. 20th. If no one implemented these then, I'll have a try. Before that, I'll implement tsm-mobilenet and supoort imgaug/albumentation in pipeline.

**Comment by innerlee** [comment]
No hurry. 

There is also lots of movie transition fx that can do some fancy temporal "mixup"

![image](https://user-images.githubusercontent.com/9464825/102436492-a8991500-4053-11eb-8263-eba72126be33.png)

**Comment by irvingzhang0512** [comment]
Find an interesting paper [here](https://arxiv.org/abs/2012.03457). This paper proposes a video augmentation strategy called `VideoMix`.

**Comment by innerlee** [comment]
haha

**Comment by irvingzhang0512** [comment]
PytorchVideo transform [random_resized_crop](https://github.com/facebookresearch/pytorchvideo/blob/dddc490ee7e7213093cbea35cc45a239f24aee2c/pytorchvideo/transforms/functional.py#L518) support `shift` mode, which looks like the movie transforms. Maybe we can support this either.
@innerlee @dreamerlin
